hibernate-core/src/main/java/org/hibernate/query/criteria/internal/predicate/InPredicate.java

```diff
@@ -197,14 +201,51 @@ public String render(boolean isNegated, RenderingContext renderingContext) {
 				}
 			}
 			else {
+				Expression lastValue = null;
 				buffer.append( '(' );
 				String sep = "";
 				for ( Expression value : values) {
+					lastValue = value;
 					buffer.append( sep )
 							.append( ( (Renderable) value )
 									.render( renderingContext ) );
 					sep = ", ";
 				}
+
+
+				if (renderingContext.getCriteriaLiteralHandlingMode() == LiteralHandlingMode.BIND &&
+						lastValue instanceof LiteralExpression) {
+
+					int bindValueCount = values.size();
+					int bindValueMaxCount = bindValueCount;
+
+					final SessionFactoryImplementor sfi = criteriaBuilder().getEntityManagerFactory().unwrap( SessionFactoryImplementor.class );
+					final Dialect dialect = sfi.getServiceRegistry().getService(JdbcServices.class).getJdbcEnvironment().getDialect();
+					final int inExprLimit = dialect.getInExpressionCountLimit();
```

Comment:
Could we move the above statements within the following if block:
```
if ( inclauseParameterPaddingEnabled ) {
    .. ...
}
```
Seems `inExprLimit` is only needed within the block and we can avoid unnecessary performance cost if padding is not enabled.